seahorn/aws-c-common-stubs/s_update_template_size_override.c

```diff
@@ -0,0 +1,55 @@
+/**
+ * FUNCTION: s_update_template_size
+ *
+ * Seahorn does not support double/float semantics;
+ * This function overrides the original implementation of the
+ * s_update_template_size function from hash_table.c by replacing floating
+ * point multiplication with nd_size_t;
+ * max_load = min { max_load_factor * size, size - 1 }
+ * so its ok to just set it to a non-det int such that:
+ * entry_count <= max_load < size
+ */
+
+#include <aws/common/hash_table.h>
+#include <aws/common/math.h>
+#include <aws/common/private/hash_table_impl.h>
+#include <nondet.h>
+#include <seahorn/seahorn.h>
+
+#include <stdlib.h>
+
+int s_update_template_size(struct hash_table_state *template,
+                           size_t expected_elements) {
+  size_t min_size = expected_elements;
+
+  if (min_size < 2) {
+    min_size = 2;
+  }
+
+  /* size is always a power of 2 */
+  size_t size;
+  if (aws_round_up_to_power_of_two(min_size, &size)) {
+    return AWS_OP_ERR;
+  }
+
+  /* Update the template once we've calculated everything successfully */
```

Comment:
use the same `ifdef` as before to allow conditional enable/disable of code instead of comments